x-pack/platform/packages/shared/onechat/onechat-common/agents/events.ts

```diff
@@ -26,8 +26,9 @@ export type ChatAgentEventBase<
 // Tool call
 
 export interface ToolCallEventData {
-  toolCallId: string;
-  toolId: StructuredToolIdentifier;
+  tool_call_id: string;
+  tool_id: PlainIdToolIdentifier;
+  tool_type: string;
```

Comment:
This is a cosmetic concern, but it feels like having a `tool` object would be useful to group `tool_call_id`, `tool_id` and now `tool_type` under a single object? This would also avoid having `type` and `tool_type` at the same level.

Reply:
If we wanted to regroup attributes, the most logical groupings would be:


```js
{
  type: 'tool_call',
  tool: {
    id: 'list_indices',
    type: 'built_in',
  },
  call: {
    id: 'call_gLpddLDrUWHA7cbUvzaO1u4g',
    args: {},
    result: '[tool result]',
  },
}
```

so I can absolutely do that, even if I'm not sure how better this really is